### PR TITLE
Add markdown editor example

### PR DESCRIPTION
**Pull request by richard-to**
While working on the LLM Rewriter, it occurred to me it would be pretty easy to make a markdown editor using Mesop. Thought it would make a decent example in the vein of the typical todo list type example.

Screenshot 2024-05-08 at 9 20 15 PM

### CODE REVIEW

**Review by wwwillchen — APPROVED** [review]
Neat! Only suggestion is to move this to the demo/ folder so it's easier to deploy on our new demo playground site (& you can still run it locally)

**Comment by richard-to** [comment]
Ok, added it to the demo folder. Wasn't sure if I did it right though.

File: mesop/examples/__init__.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from demo import main as main
+from demo import markdown_editor as markdown_editor
```
Comment (wwwillchen) [inline mesop/examples/__init__.py:2]: you don't need to import it here

Comment (wwwillchen) [reply mesop/examples/__init__.py:2]: Instead import it in https://github.com/google/mesop/blob/d132af54d52d7155fca569aa48ec64c3f2968e81/demo/main.py#L22

And then add it as an Example inside one of the Section, e.g. https://github.com/google/mesop/blob/d132af54d52d7155fca569aa48ec64c3f2968e81/demo/main.py#L77 and then it'll show up in the side menu. The embed should work then.

**Review by wwwillchen — COMMENTED** [review]
added a comment on how to add it to the demo playground

File: mesop/examples/__init__.py (hunk 1 above)
Comment (richard-to) [reply mesop/examples/__init__.py]: Thanks. That worked.

A few side notes:

- For the markdown editor, the UI gets a bit squished in the demo preview view on a laptop. Could be nice to have a way to hide the code view. Or do a "full screen" embed view.
- The hot reloading doesn't seem to track the changes in demo folder. I'm forced to stop and start the CLI. I think execute_module.py would need updates to handle it. Although I think maybe you have related hot reloading bug already in Github for this. Haven't checked.

**Pull request merged**